spec/system/purchase_system_spec.rb

```diff
@@ -170,6 +170,30 @@
           expect(Purchase.last.line_items.first.quantity).to eq(16)
         end
 
+        # Issue 2318 -- adding in tests to confirm that the category sum checks work properly.
+        # In support of change to add period supply category for annual reports
+
+        it "should allow all zero category totals" do
+          select Vendor.first.business_name, from: "purchase_vendor_id"
+          select StorageLocation.first.name, from: "purchase_storage_location_id"
+          fill_in "purchase_amount_spent", with: "10"
+          expect do
+            click_button "Save"
+          end.to change { Purchase.count }.by(1)
+        end
+        it "should save if the category totals sum to the grand total" do
+          select Vendor.first.business_name, from: "purchase_vendor_id"
+          select StorageLocation.first.name, from: "purchase_storage_location_id"
+          fill_in "purchase_amount_spent", with: "10"
+          fill_in "purchase_amount_spent_on_diapers", with: "1"
+          fill_in "purchase_amount_spent_on_adult_incontinence", with: "2"
+          fill_in "purchase_amount_spent_on_period_supplies", with: "3"
+          fill_in "purchase_amount_spent_on_other", with: "4"
+          expect do
+            click_button "Save"
+          end.to change { Purchase.count }.by(1)
+        end
```

Comment:
Does this need to be a system spec? Can't it be a model spec?

Reply:
Maybe.  I think that one was already there - I just modified it.   There might already be something for that in the model as well.  I'll take a look.

Reply:
I'm obviously misremembering!

Reply:
Ok.  It's covered off in the model already.  I will delete these two  from the system tests.

Reply:
Addressed